[macOS] Remove Big Sur specific configuration

litherum opened the pull request:
#### f0638ea74cdac6eb4bac3e0f801465c767f74e26
<pre>
[macOS] Remove Big Sur specific configuration
<a href="https://bugs.webkit.org/show_bug.cgi?id=258397">https://bugs.webkit.org/show_bug.cgi?id=258397</a>
rdar://111150554

Reviewed by Alexey Proskuryakov.

All supported macOSes are later than Big Sur.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:

Canonical link: <a href="https://commits.webkit.org/265433@main">https://commits.webkit.org/265433@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2743f2a8832e79d99e8892f068874b48dc61f5f6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10873 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11089 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11397 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12521 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10409 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13463 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11068 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13320 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11032 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11935 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12925 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9228 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9818 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9209 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10299 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9970 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13213 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10315 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10427 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8510 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10983 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9592 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2997 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2609 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13865 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11291 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/10288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2761 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -139,7 +139,7 @@
 
 #endif // !PLATFORM(MAC) && !PLATFORM(MACCATALYST) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
 
-#if !defined(ENABLE_APPLE_PAY_SESSION_V11) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300) || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140500) || PLATFORM(VISION))
```
Not sure what your plan of attack is, but the iOS check is also unneeded.

litherum replied:
Yep. The plan of attack is to do different platforms in different patches.

litherum replied:
(The new plan of attack is to do this year-by-year instead of year-by-year and platform-by-platform. Will start that after landing this current batch of patches.)

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -199,7 +199,7 @@
 #endif
 
-#if !defined(ENABLE_CFPREFS_DIRECT_MODE) && (PLATFORM(IOS_FAMILY) || PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000)
+#if !defined(ENABLE_CFPREFS_DIRECT_MODE) && (PLATFORM(IOS_FAMILY) || PLATFORM(MAC))
```
`PLATFORM(IOS_FAMILY) || PLATFORM(MAC)` is just `PLATFORM(COCOA)`

litherum replied:
Good point!

litherum replied:
I suppose `PLATFORM(COCOA)` isn't necessary in `PlatformEnableCocoa.h` so I guess I'll just remove it and just do `#if !defined(ENABLE_CFPREFS_DIRECT_MODE)`

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -445,7 +445,7 @@
 #endif
 
-#if !defined(ENABLE_MEDIA_USAGE) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140000))
+#if !defined(ENABLE_MEDIA_USAGE) && (PLATFORM(MAC) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140000))
```
More unnecessary iOS version checking

litherum replied:
Yep, will fix this in a follow-up.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -765,15 +765,15 @@
 #endif
 
-#if !defined(ENABLE_VORBIS) && (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300 || PLATFORM(IOS) || PLATFORM(IOS_SIMULATOR) || PLATFORM(VISION))
+#if !defined(ENABLE_VORBIS) && (PLATFORM(MAC) || PLATFORM(IOS) || PLATFORM(IOS_SIMULATOR) || PLATFORM(VISION))
```
`PLATFORM(IOS) || PLATFORM(IOS_SIMULATOR)` is superfluous IIRC.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -958,7 +954,7 @@
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000 && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300) \
+#if PLATFORM(MAC) \
     || (PLATFORM(IOS_FAMILY) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140500) \
     || (PLATFORM(APPLETV) && __TV_OS_VERSION_MAX_ALLOWED >= 140500) \
     || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MAX_ALLOWED >= 70400)
```
:(

litherum replied:
Is the frown because I only touched Mac and not the other platforms?

aproskuryakov replied:
Yes.

litherum replied:
Okay, will address this in a follow-up.

litherum committed:
[macOS] Remove Big Sur specific configuration
https://bugs.webkit.org/show_bug.cgi?id=258397
rdar://111150554

Reviewed by Alexey Proskuryakov.

All supported macOSes are later than Big Sur.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:

Canonical link: https://commits.webkit.org/265433@main
